rpc/params/caas.go

```diff
@@ -49,6 +49,7 @@ type CAASApplicationProvisioningInfo struct {
 	APIAddresses         []string                     `json:"api-addresses"`
 	CACert               string                       `json:"ca-cert"`
 	Constraints          constraints.Value            `json:"constraints"`
+	CharmConstraints     constraints.CharmValue       `json:"charmconstraints"`
```

Comment:
Using `constraints.Value` above was wrong of us. Define a new struct here in this package.

Reply:
We don't want to refactor existing code. But we don't want to continue with a poor implementation choice for new code.